app/src/main/java/org/zephyrsoft/trackworktime/timer/TimeCalculatorV2.java

```diff
@@ -138,17 +144,24 @@ public boolean containsEvents() {
 	private long currentBalance = 0;
 	private int futureWorkDays = -1;
 
-	public TimeCalculatorV2(DAO dao, TimerManager timerManager, LocalDate startDate, boolean handleFlexiTime) {
+	public TimeCalculatorV2(DAO dao, TimerManager timerManager, LocalDate startDate, SharedPreferences preferences) {
 		this.dao = dao;
 		this.timerManager = timerManager;
 
-		this.handleFlexiTime = handleFlexiTime;
+		this.handleFlexiTime = preferences.getBoolean(Key.ENABLE_FLEXI_TIME.getName(), false);
 		if (handleFlexiTime) {
 			this.flexiReset = timerManager.getFlexiReset();
 		} else {
 			this.flexiReset = FlexiReset.NONE;
 		}
 
+		this.roundingEnabled = preferences.getBoolean(Key.ROUNDING_ENABLED.getName(), false);
+		if (preferences.getBoolean(Key.ROUNDING_ENABLED.getName(), false)) {
+			this.smallestTimeUnit = Integer.parseInt(preferences.getString(Key.SMALLEST_TIME_UNIT.getName(), "1"));
+		} else {
+			this.smallestTimeUnit = 1;
```

Comment:
The `else` part can be left out if the variable is initialized with the value `1` above:
```
private final int smallestTimeUnit = 1;
```

Reply:
Maybe I am missing something, but I get an error if I do that.

Reply:
Sorry, my bad - the `final` keyword is the problem. With
```
private int smallestTimeUnit = 1;
```
it works.